File: primary_collections/groups/edit_one_groups.rb
#!/usr/bin/env ruby

require 'json'
require 'net/http'
require 'openssl'
require 'uri'

uri = URI.parse("https://#{ENV['MIQ']}/api/groups/24")

http = Net::HTTP.new(uri.host, uri.port)
http.use_ssl = true
http.verify_mode = OpenSSL::SSL::VERIFY_NONE

request = Net::HTTP::Post.new(uri.request_uri)
request.basic_auth("admin", "smartvm")
request.body = '
{
  "action" : "edit",
  "resource" : {
    "description" : "updated_sample_group",
    "filters"     : {
      "belongsto" : [ "/managed/area/1", "/managed/area/2", "/managed/area/3" ],
      "managed"   : [[ "/managed/infra/1", "/managed/infra/2" ], ["/managed/other/3"]]
    }
  }
}
'
response = http.request(request)
puts JSON.pretty_generate(JSON.parse(response.body.strip))

File: primary_collections/groups/edit_one_groups.py
#!/usr/bin/env python

import requests
import json
import os

url = 'https://' + str(os.environ["MIQ"]) + '/api/groups/23'

request_body = { 'action':  'edit', 'resource' : { 'description' : 'updated group description from python' } }
headers = {'Content-type': 'application/json'}
response = requests.post(url, data=json.dumps(request_body), headers=headers,  auth=('admin', 'smartvm'), verify=False)
print("Result:\n" + json.dumps(json.loads(response.text), indent=4, sort_keys=True))
